packages/marshaller/src/dashy.ts

```diff
@@ -121,6 +122,21 @@ export class Dashy extends SplitPanel {
     restore(json: DDL2.Schema): Promise<void> {
         this._lhsSheet.active(this._lhsDashboard);
         return this.clear().then(() => {
+            if (json && json.properties && json.properties.layout instanceof Array) {
+                this._lhsDashboard.widget().target(null);
+                this._lhsDashboard._dockPanel = new DashboardGrid(this._elementContainer)
+                    .on("vizActivation", (elem: Element) => {
+                        this._lhsDashboard.vizActivation(elem);
+                    })
+                    ;
+                this._lhsDebugClone._dockPanel = new DashboardGrid(this._lhsDebugCloneEC)
+                    .on("vizActivation", (elem: Element) => {
+                        this._lhsDashboard.vizActivation(elem);
+                    })
+                    ;
+                this._lhsDashboard.widget(this._lhsDashboard._dockPanel as any);
+                this._lhsDebugClone.widget(this._lhsDebugClone._dockPanel as any);
+            }
```

Comment:
This code looks like it belongs in dashboard and that the "clone" work should not happen until clone is actually being rendered?